src/main.rs

```diff
@@ -159,16 +173,13 @@ mod cmd {
             "Repositories must be in the format <group>-proto-<stability>"
         );
 
-        ensure!(
-            repository.contains("-proto-"),
-            "Only proto repositories are allowed"
-        );
-
         let (package, version) = dependency
             .split_once('@')
             .wrap_err("Invalid dependency specification")?;
 
-        let package = package.parse::<PackageId>()?;
+        let package = package
+            .parse::<PackageId>()
+            .wrap_err("Invalid package id supplied")?;
```

Comment:
In my experience, the context is nice, but it's even nicer with `.wrap_err_with(|| format!("Invalid package id supplied: {package}"))`!